extendr-api/tests/extendr_macro.rs

```diff
@@ -0,0 +1,109 @@
+use extendr_api::prelude::*;
+use extendr_macros::*;
+
+#[extendr(use_try_from = true)]
+fn test_i32(val: i32) -> i32 {
+    val
+}
+
+#[extendr(use_try_from = true)]
+fn test_i16(val: i16) -> i16 {
+    val
+}
+
+#[extendr(use_try_from = true)]
+fn test_option_i32(val: Option<i32>) -> i32 {
+    if let Some(i) = val {
+        i
+    } else {
+        -1
+    }
+}
+
+#[extendr(use_try_from = true)]
+fn test_option_f64(val: Option<f64>) -> f64 {
+    if let Some(i) = val {
+        i
+    } else {
+        -1.0
+    }
+}
+
+#[extendr(use_try_from = true)]
+fn test_option_i16(val: Option<i16>) -> i16 {
+    if let Some(i) = val {
+        i
+    } else {
+        -1
+    }
+}
+
+#[test]
+fn happy_path() {
+    unsafe {
+        test! {
+            // Matching integer.
+            assert_eq!(new_owned(wrap__test_i32(r!(1).get())), r!(1));
+
+            // i32 takes any numeric.
+            assert_eq!(new_owned(wrap__test_i32(r!(1.0).get())), r!(1));
```

Comment:
I am not sure what 'takes any numeric' means here. What will happen if we pass `1.5`?
What would be the behavior?
If the value is somehow truncated/converted to integer space, this may be an issue, and extremely hard to debug.

Same applies to `Option<i32>`

If this is OK, may I then suggest to add another option, like `#[extendr(use_strict_types = true)]` or `#[extendr(vctrs_compat = true)]` to enable strict type checks?

Reply:
If you feel like doing so, we should replace the "as" in `impl_try_from_scalar!` with TryFrom.
This would also range-check small numbers.

I'm going to close this ASAP so that more PRs can follow.

Reply:
We now have a template for adding other options, so suggestions like this are welcome.

Reply:
Just wanted to point out that I think strict type checks should be the default, not optional. If somebody writes `i32` as their input type we shouldn't silently round from 1.5 to 1, we should throw a runtime error.

Reply:
I fully support this. My preferred solution would be something similar to {vctrs} behavior, which should at least be available as an option.